pyanaconda/ui/gui/spokes/welcome.py

```diff
@@ -105,22 +105,25 @@ def _apply_selected_locale(self, locale):
 
     def _apply_geolocation_results(self):
         """Apply the geolocation results if any."""
+        # Without timezone module, we have neither geolocation results nor way to apply them.
+        if not self._tz_module:
+            return
+
+        geoloc_result = GeolocationData.from_structure(self._tz_module.GeolocationResult)
+
         # Skip timezone and keyboard default setting for kickstart installs.
         # The user may have provided these values via kickstart and if not, we
         # need to prompt for them. But do continue if geolocation-with-kickstart
         # is enabled.
-        if flags.flags.automatedInstall and not geoloc.geoloc.enabled:
-            return
-
-        if not self._tz_module:
+        if flags.flags.automatedInstall and geoloc_result.is_empty():
             return
 
         loc_timezones = localization.get_locale_timezones(self._l12_module.Language)
-        if geoloc.geoloc.result.timezone:
+        if geoloc_result.timezone:
```

Comment:
Btw. I think that `elif loc_timezones and not self._tz_module.Timezone:` is never used, because the timezone is always set. Since there are no bug reports for that, I recommend to remove this part of the code.